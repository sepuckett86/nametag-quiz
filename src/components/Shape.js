import React from 'react';
import PropTypes from 'prop-types';

function Shape({ text, color, backgroundColor }) {
  const fontStyles = {
    color
  };

  const backgroundStyles = {
    backgroundColor,
    width: '100px',
    height: '100px',
    border: '2px black solid'
  };

  return (
    <>
    <div style={backgroundStyles}>
      <p style={fontStyles}>{text}</p>
    </div>
    </>
  );
}

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired
};

export default Shape;
