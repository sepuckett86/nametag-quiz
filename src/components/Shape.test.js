import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders App', () => {
    const wrapper = shallow(
      <Shape 
        color="#000000"
        backgroundColor="#000000"
        text="Some text here"
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
