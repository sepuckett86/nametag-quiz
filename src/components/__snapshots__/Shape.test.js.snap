// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders App 1`] = `
<Fragment>
  <div
    style={
      Object {
        "backgroundColor": "#000000",
        "border": "2px black solid",
        "height": "100px",
        "width": "100px",
      }
    }
  >
    <p
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Some text here
    </p>
  </div>
</Fragment>
`;
